salt/modules/mount.py

```diff
@@ -483,15 +487,15 @@ def rm_fstab(name, device, config='/etc/fstab'):
 
     except (IOError, OSError) as exc:
         msg = "Couldn't read from {0}: {1}"
-        raise CommandExecutionError(msg.format(config, str(exc)))
+        raise CommandExecutionError(msg.format(config, six.text_type(exc)))
 
     if modified:
         try:
             with salt.utils.files.fopen(config, 'w+') as ofile:
-                ofile.writelines(lines)
+                ofile.writelines(salt.utils.data.decode(lines))
```

Comment:
This needs to be `encode`, not `decode`, and the `fopen()` should have a mode of `'wb'`.